content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -173,22 +175,44 @@ class CreateAndIssueMarsVoucherInitiator @JsonConstructor constructor(private va
-4. Add exceptions for when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
-5. Insert a method for finding the notary.
+4. Add appropriate error handling - exceptions must be thrown when required fields (`voucherDesc`, `holder`, `recipientParty`) are not found.
+5. Insert the following method for finding the notary: `notaryLookup.getNotary(CordaX500Name.parse("O=notary, L=London, C=GB"))!!`.
+
+{{< note >}}
+
+In Corda 5 the notary is predefined. You will be "spending" your state with a particular notary tieing that state to it (notary change excepted). Thus, you cannot just pick the first one as the list order may change, especially in a multi-network or app environment where different notaries are whitelisted for different apps. You can find the full status of the notary by starting the network and printing out its status.
+
+{{< /note >}}
+
 6. Build the output `MarsVoucher` state with a `UniqueIdentifier`.
-7. Build the transaction using a `txCommand` and the `transactionBuilderFactory` service you added when injecting services.
-8. Verify that the transaction is valid using the `txBuilder.verify` method.
-9. Sign the transaction using the `txBuilder.sign` method.
-10. Send the state to the counterparty and receive it back with their signature. Use the `flowMessaging` service to send the state to the `recipientParty`. Use the `flowEngine` service with the subflow `CollectSignaturesFlow` to get the signature of the counterparty.
-11. Notarize and record the transaction in both parties' vaults using `FinalityFlow` with a `fullySignedTx`.
-12. Return a `SignedTransactionDigest`. Use the `jsonMarshallingService` to parse the transaction's output. Include the transaction's output states, ID, and signatures.
+
+##### Build the transaction
+
+Your transaction is a proposal to the ledger that the counterparty must agree to. You must ensure that the proposal adheres to the rules of the smart contract and you must sign it before sending it to the counterparty. Also, ensure your transaction is valid from a business perspective. The proposal here is that you'll create a voucher for a trip to Mars and then issue that voucher to a notary and a designated party - Peter.
+
+1. Build the transaction using a `txCommand` and the `transactionBuilderFactory` service you added when injecting services.
+2. Verify that the transaction is valid using the `txBuilder.verify` method.
+3. Sign the transaction using the `txBuilder.sign` method.
+   Once the transaction is signed, you cannot remove a signature from it.
+
+##### Interact with the counterparty
+
+This is where you send the proposal to Peter. When consensus that the transaction's elements are unique and understood is achieved, the transaction is notarized by the notary and stored in both parties' vaults.
```

Comment:
"When consensus that the transaction's elements are unique and understood is achieved", this part doesn't quite make sense. Not sure what it should say though.